### ISSUE TITLE
Members and 2.6.4

### ISSUE THREAD

**Issue by andrewminton** [symphonycms/members#272]
I keep getting an error:

Fatal error: Call to a member function canProcessEvent() on a non-object in /var/www/html/extensions/members/extension.driver.php on line 1168

I'm using latest master of members and symphony 2.6.4 pulled from github.

I have a section in the backend called users that is attached to the members section, but I'm trying to submit a simple front end form that isn't attached to the members section but to a different section all together. But this error shows upon submitting the form.

It's related to permissions, but do I really need permissions on a section that isn't attached to members??

**Comment by andrewminton** [comment]
Hmmm, seems that the 'Users' section has a role of Public. and if it exists, no one can submit a form unless they are logged in. That seems odd to me.

**Comment by michael-e** [comment]
You **must** set event permissions. The default is "no permission" (for good reason).

But: Members should not throw a fatal error. This is a bug.

**Comment by andrewminton** [comment]
Permissions are set for the event itself. 

'Create new' and 'Edit own'. But because I had a role of 'Public' attached to my users section, it really didn't like it and failed miserably.

**Comment by nitriques** [comment]
Since I do not use this extension often, I do not know how to help.

@andrewminton Can you call `debug_print_backtrace()` just before the crash ?

**Comment by andrewminton** [comment]
How do you want me t provide the output? it's a long trace.

**Comment by nitriques** [comment]
Done. Well it seems we are only missing an if statement.

```php
$success = false;
if ($role) {
  $success = $role->canProcessEvent($event_handle, $action, $required_level) ? true : false;
}
```php

**Comment by andrewminton** [comment]
Even with that condition and a Role in my Users section if still finds a Role and it still errors as before.

**Comment by nitriques** [comment]
> Rather than close this issue now, is it worth checking this issue can be replicated by others?

Let others open a new issue, it's easier to maintain.

Can you send a PR please ? :)

**Comment by nitriques** [comment]
Fixed with #273
